### PR TITLE
Disregard callback url regex validation when accessurl is configured in the app

### PR DESCRIPTION
**Pull request by ashanthamara**
Merge before https://github.com/wso2/identity-apps/pull/4945

### Related Issue
- https://github.com/wso2/product-is/issues/17999

### Related PRs
- https://github.com/wso2/identity-apps/pull/4945
- https://github.com/wso2-extensions/identity-event-handler-notification/pull/212

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java [hunk 1]
```diff
@@ -719,6 +719,28 @@ public static String getCallbackURL(org.wso2.carbon.identity.recovery.model.Prop
...
+            }
+        }
+
+        if (StringUtils.isNotBlank(accessURL)) {
+            URI uri = new URI(accessURL);
+            accessURL = new URI(uri.getScheme(), uri.getAuthority(), uri.getPath(), null, null)
+                    .toString();
+        }
+        return accessURL;
+    }
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java:742]: shall we use Optional here?

Comment (ashanthamara) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java]: Changed accessUrl property to isAccessUrlAvailable boolean property by https://github.com/wso2-extensions/identity-governance/pull/787/commits/47e973bf1bd8817ab28abe25549992ec43e49d2b. Hence no need to use optional there

**Review by chamathns — APPROVED** [review]

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java [hunk 2]
```diff
@@ -719,6 +719,24 @@ public static String getCallbackURL(org.wso2.carbon.identity.recovery.model.Prop
         return callbackURL;
     }
 
+    /**
+     * Get isAccessUrlAvailable property value.
+     *
+     * @param properties Properties array.
+     * @return Boolean value of the isAccessUrlAvailable property.
+     */
+    public static Boolean isAccessUrlAvailable(org.wso2.carbon.identity.recovery.model.Property[] properties) {
```
Comment (inthirakumaaran) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java:728]: Change the Boolean to boolean here

Comment (ashanthamara) [reply components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java:728]: Fixed with https://github.com/wso2-extensions/identity-governance/pull/815

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java [hunk 3]
```diff
@@ -719,6 +719,24 @@ public static String getCallbackURL(org.wso2.carbon.identity.recovery.model.Prop
...
+
+        if (properties != null) {
```
Comment (Yoshani) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java:730]: Shall we have an early return for null check?
```
if (properties == null) {
        return false;
 }
```

**Pull request merged**